packages/fuels-core/src/types/wrappers/transaction.rs

```diff
@@ -325,6 +326,20 @@ macro_rules! impl_tx_wrapper {
             }
         }
 
+        impl TryFrom<FuelTransaction> for $wrapper {
+            type Error = Error;
+
+            fn try_from(tx: FuelTransaction) -> Result<Self> {
+                match tx {
+                    FuelTransaction::$wrapped(tx) => Ok(tx.into()),
+                    _ => Err(error!(
```

Comment:
Had to `impl TryFrom<FuelTransaction> for Transaction`, where logic depends on the enum variant being named exactly the same as the transaction type being wrapped, e.g., `FuelTransaction::Script(tx: Script)`. Any other way to do a round-trip `T -> fuel_tx::Transaction -> T where T: Transaction` would be very appreciated.

Reply:
We will probably have to re-think the whole logic and check if we can simplify this. For this PR the `try_into` is ok